lib/rpc-client.js

```diff
@@ -0,0 +1,60 @@
+import {
+  FilecoinRpcError,
+  INVALID_RPC_RESPONSE,
+  NO_RESULT_IN_RESPONSE,
+  RPC_ERROR,
+} from './errors.js'
+
+/**
+ * @param {string} method
+ * @param {unknown[]} params
+ * @param {string} rpcUrl
+ * @param {string} rpcAuth
+ * @param {object} [options]
+ * @param {globalThis.fetch} [options.fetch]
+ */
+export async function rpc(method, params, rpcUrl, rpcAuth, { fetch = globalThis.fetch } = {}) {
+  const req = new Request(rpcUrl, {
+    method: 'POST',
+    headers: {
+      'content-type': 'application/json',
+      accepts: 'application/json',
+      authorization: `Bearer ${rpcAuth}`,
```

Comment:
rpcAuth should be optional (not all RPC API providers require it). When rpcAuth is not set, `authorization` header should be omitted from the request.